File: android/JavaEmptyProject/app/src/main/java/com/lynx/javaemptyproject/LynxActivity.java
package com.lynx.javaemptyproject;

import android.content.Context;
import android.content.Intent;
import android.os.Bundle;

import androidx.appcompat.app.AppCompatActivity;

public class LynxActivity extends AppCompatActivity {

    private static final String TAG = "LynxActivity";
    private static final String INTENT_EXTRA_URL = "url";

    public static void start(Context context, String url) {
        Intent intent = new Intent(context, LynxActivity.class);
        intent.putExtra(INTENT_EXTRA_URL, url);
        context.startActivity(intent);
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        render(getIntent().getStringExtra(INTENT_EXTRA_URL));
    }

    private void render(String url) {
        // TODO: render lynx view here
    }
}

File: android/KotlinEmptyProject/app/src/main/java/com/lynx/kotlinemptyproject/LynxActivity.kt
package com.lynx.kotlinemptyproject

import android.content.Context
import android.content.Intent
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity

class LynxActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        render(intent.getStringExtra(INTENT_EXTRA_URL))
    }

    private fun render(url: String?) {
        // TODO: render lynx view here
    }

    companion object {
        private const val TAG = "LynxActivity"
        private const val INTENT_EXTRA_URL = "url"

        fun start(context: Context, url: String) {
            val intent = Intent(context, LynxActivity::class.java)
            intent.putExtra(INTENT_EXTRA_URL, url)
            context.startActivity(intent)
        }

    }
}